core/unit_test/TestRangePolicyConstructors.hpp

```diff
@@ -121,4 +122,82 @@ TEST(TEST_CATEGORY_DEATH, range_policy_invalid_bounds) {
 #endif
 }
 
+TEST(TEST_CATEGORY_DEATH, range_policy_implicitly_converted_bounds) {
+  using UIntIndexType = Kokkos::IndexType<unsigned>;
+  using IntIndexType  = Kokkos::IndexType<int>;
+  using UIntPolicy    = Kokkos::RangePolicy<TEST_EXECSPACE, UIntIndexType>;
+  using IntPolicy     = Kokkos::RangePolicy<TEST_EXECSPACE, IntIndexType>;
+
+#ifndef KOKKOS_ENABLE_DEBUG
+  GTEST_SKIP();
+#endif
```

Comment:
Just drop this and the guard in the source file and (assuming that the tests are passing) I will approve.

Reply:
Removed the ifdef guard and gtest_skip.